homeassistant/components/netatmo/light.py

```diff
@@ -32,61 +30,56 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the Netatmo camera light platform."""
-    data_handler = hass.data[DOMAIN][entry.entry_id][DATA_HANDLER]
-    data_class = data_handler.data.get(CAMERA_DATA_CLASS_NAME)
-
-    if not data_class or data_class.raw_data == {}:
-        raise PlatformNotReady
-
-    all_cameras = []
-    for home in data_handler.data[CAMERA_DATA_CLASS_NAME].cameras.values():
-        for camera in home.values():
-            all_cameras.append(camera)
-
-    entities = [
-        NetatmoLight(
-            data_handler,
-            camera["id"],
-            camera["type"],
-            camera["home_id"],
-        )
-        for camera in all_cameras
-        if camera["type"] == "NOC"
-    ]
 
-    _LOGGER.debug("Adding camera lights %s", entities)
-    async_add_entities(entities, True)
+    @callback
+    def _create_camera_light_entity(netatmo_device: NetatmoDevice) -> None:
+        try:
+            getattr(netatmo_device.device, "floodlight")
+        except AttributeError:
+            return
```

Comment:
You can use hasattr, then you don't need the try/except
```suggestion
        if not hasattr(netatmo_device.device, "floodlight"):
            return
```